packages/vm/src/static_analysis.rs

```diff
@@ -0,0 +1,204 @@
+use parity_wasm::elements::{deserialize_buffer, Internal, Module};
+use std::collections::HashSet;
+
+use crate::errors::{VmError, VmResult};
+
+pub const REQUIRED_IBC_EXPORTS: &[&str] = &[
+    "ibc_channel_open",
+    "ibc_channel_connect",
+    "ibc_channel_close",
+    "ibc_packet_receive",
+    "ibc_packet_ack",
+    "ibc_packet_timeout",
+];
+
+pub fn deserialize_wasm(wasm_code: &[u8]) -> VmResult<Module> {
+    deserialize_buffer(&wasm_code).map_err(|err| {
+        VmError::static_validation_err(format!(
+            "Wasm bytecode could not be deserialized. Deserialization error: \"{}\"",
+            err
+        ))
+    })
+}
+
+pub fn exported_functions(module: &Module) -> HashSet<String> {
+    module
+        .export_section()
+        .map_or(HashSet::default(), |export_section| {
+            export_section
+                .entries()
+                .iter()
+                .filter_map(|entry| match entry.internal() {
+                    Internal::Function(_) => Some(entry.field().to_string()),
+                    _ => None,
+                })
+                .collect()
+        })
+}
+
+/// Returns true if and only if all IBC entry points ([`REQUIRED_IBC_EXPORTS`])
+/// exist as exported functions. This does not guarantee the entry points
+/// are functional and for simplicity does not even check their signatures.
+pub fn has_ibc_entry_points(module: &Module) -> bool {
+    let available_exports = exported_functions(module);
+    let mut ok = true;
+    for required_export in REQUIRED_IBC_EXPORTS {
+        if !available_exports.contains(*required_export) {
+            ok = false;
```

Comment:
`break;`

Reply:
I thought about this as well, but it is only 6 elements.

We could remove the `ok` variable completely by returning early from the function 🤔

Reply:
Yeah, I guess `return false` would be cleaner.
Not important for efficiency, but possible cleanup pr

Reply:
But what we really want is https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.all